### PR TITLE
libnghttp2_asio: Added access to a requests remote endpoint

### PR DESCRIPTION
**Pull request by apohl79**

### CODE REVIEW

File: src/asio_server_http2_handler.cc [hunk 1]
```diff
@@ -449,6 +453,8 @@ response *http2_handler::push_promise(boost::system::error_code &ec,
 
 boost::asio::io_service &http2_handler::io_service() { return io_service_; }
 
+boost::asio::ip::tcp::endpoint http2_handler::remote_endpoint() { return remote_ep_; }
```
Comment (tatsuhiro-t) [inline src/asio_server_http2_handler.cc]: I think we can return const reference of remote_ep_ here (and all delegating functions).

**Comment by tatsuhiro-t** [comment]
Thank you for PR!  Could you check my comment inline?

File: src/asio_server_http2_handler.cc (hunk 1 above)
Comment (apohl79) [inline src/asio_server_http2_handler.cc]: You are right, fixed and pushed, thx!

**Comment by tatsuhiro-t** [comment]
Thank you!  Merged now.

**Pull request merged**